content/articles/advanced-concepts-of-programming-using-dart/index.md

```diff
@@ -0,0 +1,370 @@
+# Advanced Concepts of Programming using Dart 
+	
+In this tutorial, we will learn to appreciate some advanced concepts of programming using Dart a language that can be both robust and rigid to use.
+
+![advanced-concepts](/engineering-education/advanced-concepts-of-programming-using-dart/advanced-concepts.jpg)
+
+### Introduction
+Programming is a way of writing syntaxes that instructs a computer to perform certain tasks the way we want it to in order to achieve a specific task. For one to achieve programming, certain concepts like Function declaration and usage, conditional situation handling, variable declaration - assignment - and reassignment, looping e.t.c. provide a way to achieve programming. These concepts listed can be classified as basic programming concepts and are essential knowledge needed for other advanced concepts that exist within programming. We will be discussing some of those advanced concepts that make programming a whole lot better and more fun to use. 
+
+ 
+In this tutorial, we will:
+- Write codes of situations that will help us understand advanced concepts of programming.
+- Learn about advanced concepts of programming such as classes, objects, abstraction, polymorphism, encapsulation, inheritance, Access modifiers, static classes,
+- Learn the concepts using dart as the language of choice. But the concepts are the same for any programming language we can think of only that they are represented using different syntaxes.
+ 
+### Prerequisite
+The concepts discussed here are not for beginners and as such, It is important that we have a basic understanding of programming and basic programming concepts in any programming language such as Java, Python, or JavaScript.
```

Comment:
List down the prerequisites instead of having them as a paragraph i.e
- Basic understanding of something.
- Something installed.